pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -91,6 +91,7 @@ public void initialize() {
             Reader<byte[]> reader = pulsarClient.newReader()
                     .topic(this.workerConfig.getFunctionMetadataTopic())
                     .startMessageId(MessageId.earliest)
+                    .readerName(workerConfig.getWorkerId() + "-function-metadata-manager")
```

Comment:
I don't know if its needed to add "-function-metadata-manager".  I would say that this information as a property e.g. component: function-metadata-manager but properties is not exposed for the reader

Reply:
I would rather prefer it to be more specific than just assumed.

Reply:
I meant to add it here

```
can you also add

.subscriptionRolePrefix(workerConfig.getWorkerId())
```